File: ClientApp/app/models/palinitem.ts
interface IPalinItem
{
    Id: number;
    Text: string;
    Hash: string;
}

export class PalinItem
{
    public Id: number = 0;
    public Text: string = "";
    public Hash: string = "";

    contructor(item: IPalinItem) {
        this.Id = item.Id;
        this.Text = item.Text;
        this.Hash = item.Hash;    
    }
}

File: Models/PalinItem.cs
using System.ComponentModel.DataAnnotations;

namespace Palindrome.Models
{
    public class PalinItem
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public string Text { get; set; }
        [Required]
        public string Hash { get; set; }
    }
}